naga/src/proc/constant_evaluator.rs

```diff
@@ -254,6 +254,256 @@ gen_component_wise_extractor! {
     ],
 }
 
+/// Vector for each [`Literal`] type
+///
+/// This type ensures that all elements have same type
+#[derive(Debug)]
+enum LiteralVector {
+    F64(ArrayVec<f64, { crate::VectorSize::MAX }>),
+    F32(ArrayVec<f32, { crate::VectorSize::MAX }>),
+    U32(ArrayVec<u32, { crate::VectorSize::MAX }>),
+    I32(ArrayVec<i32, { crate::VectorSize::MAX }>),
+    U64(ArrayVec<u64, { crate::VectorSize::MAX }>),
+    I64(ArrayVec<i64, { crate::VectorSize::MAX }>),
+    Bool(ArrayVec<bool, { crate::VectorSize::MAX }>),
+    AbstractInt(ArrayVec<i64, { crate::VectorSize::MAX }>),
+    AbstractFloat(ArrayVec<f64, { crate::VectorSize::MAX }>),
+}
+
+impl LiteralVector {
+    #[allow(clippy::pattern_type_mismatch, clippy::missing_const_for_fn)]
```

Comment:
nitpick: I think @jimblandy would be sad if I accepted this `clippy::pattern_type_mismatch` trigger, so let's not avoid it. This feedback applies to other places where this lint is being fired.

suggestion(blocking): Adding `const` seems a harmless resolution to the `clippy::missing_const_for_fn` lint. Let's do that everywhere we can, instead of suppressing the lint.